source/_posts/tips-for-beginners.md

```diff
@@ -0,0 +1,253 @@
+  el: '#demo',
+
+  data: {
+    wordOfChild: ''
+  },
+
+  methods: {
+    listenToChild: function(what) {
+      return this.wordOfChild = what
+    }
+  },
+
+  components: {
+    'child-component': {
+      props: ['say'],
+      template: '<button type="button" v-on="click: say(\'Yeah!\')">Say Yeah!</button>'
+    }
+  }
+})
+```
+
+**Result:**
+
+<div id="demo"><child-component say="{{listenToChild}}"></child-component><p v-if="wordOfChild.length">Child says: {&#123;wordOfChild&#125;}</p></div>
+
+<script>
+  new Vue({
+    el: '#demo',
+
+    data: {
+      wordOfChild: ''
+    },
+
+    methods: {
+      listenToChild: function(what) {
+        return this.wordOfChild = what
+      }
+    },
+
+    components: {
+      'child-component': {
+        props: ['say'],
+        template: '<button type="button" v-on="click: say(\'Yeah!\')">Say Yeah!</button>'
+      }
+    }
+  })
+</script>
+
+Instances can also communicate using [Events](/api/instance-methods.html#Events), but this is mostly for more complex use cases, where `props` can't get the job done. In following demo, you can see how parent ViewModel broadcasts event to all children:
+
+<iframe width="100%" height="300" src="//jsfiddle.net/azamatsharapov/dxpvb3n3/embedded/" allowfullscreen="allowfullscreen" frameborder="0"></iframe>
+
+## Adding data properties
+
+<p class="tip">In ECMAScript 5 there is no way to detect when a new property is added to an Object, or when a property is deleted from an Object. To deal with that, observed objects will be augmented with three methods: $add(key, value), $set(key, value) and $delete(key). These methods can be used to add / delete properties from observed objects while triggering the desired View updates. The difference between $add and $set is that $add will return early if the key already exists on the object, so just calling obj.$add(key) won’t overwrite the existing value with undefined.</p>
+
+The quote above is taken from guide and is important to mention, because many beginners run into issues with it. If you need to add new or delete existing data properties, use augmented methods as mentioned above. Another important thing you should know is, when you change data-bound Array directly using indices, Vue.js will not pick your changes and so View will not be updated. Again, you can use augmented methods to notify Vue.js about those changes. Also, augmented methods can take path as a first argument, which is useful. See example below.
+
+``` js
+new Vue({
+  el: '#demo',
+  
+  data: {
+    arrayItems: [],
+    objectItems: {}
+  },
+
+  ready: function() {
+    // adds new item and sets 'value' of it
+    this.arrayItems.$add(0, 'someValue')
+    
+    // Head Up! This will *not* change arrayItems[0], use $set() for that
+    // arrayItems[0] will still have 'someValue' as value
+    this.arrayItems.$add(0, 'newValue')
+
+    // using path! so objectItems becomes
+    // {a: { b: 'value' } }
+    this.objectItems.$add('a.b', 'value')
```

Comment:
Hmm, $add actually doesn't support paths. The only methods that accept paths are `vm.$get` and `vm.$set`.